gensim/models/translation_matrix.py

```diff
@@ -0,0 +1,329 @@
+    """
+    def __init__(self, matrix, index2word):
+        """
+        `matrix`: N * length_of_word_vec, which store the word's vector
+        `index2word`: a list of words in the `Space` object
+        `word2index`: a dict which for word indexing
+        """
+        self.mat = matrix
+        self.index2word = index2word
+
+        # build a dict to map word to index
+        self.word2index = {}
+        for idx, word in enumerate(self.index2word):
+            self.word2index[word] = idx
+
+    @classmethod
+    def build(cls, lang_vec, lexicon=None):
+        """
+        construct a space class for the lexicon, if it's provided.
+        Args:
+            `lang_vec`: word2vec model that extract word vector for lexicon
+            `lexicon`: the default is None, if it is not provided, the lexicon
+                    is all the lang_vec's word, i.e. lang_vec.vocab.keys()
+        Returns:
+            `Space` object for the lexicon
+        """
+        # `words` to store all the word that
+        # `mat` to store all the word vector for the word in 'words' list
+        words = []
+        mat = []
+        if lexicon is not None:
+            # if the lexicon is not provided, using the all the Keyedvectors's words as default
+            for item in lexicon:
+                words.append(item)
+                mat.append(lang_vec.syn0[lang_vec.vocab[item].index])
+
+        else:
+            for item in lang_vec.vocab.keys():
+                words.append(item)
+                mat.append(lang_vec.syn0[lang_vec.vocab[item].index])
+
+        return Space(mat, words)
+
+    def normalize(self):
+        """ normalized the word vector's matrix """
+        self.mat = self.mat / np.sqrt(np.sum(np.multiply(self.mat, self.mat), axis=1, keepdims=True))
+
+
+class TranslationMatrix(utils.SaveLoad):
+    """
+    Objects of this class realize the translation matrix which map the source language
+    to the target language.
+     The main methods are:
+
+    1. constructor,
+    2. the `train` method, which initialize everything needed to build a translation matrix
+    3. the `translate` method, which given new word and its vector representation,
+    we map it to the other language space by computing z = Wx, then return the
+    word whose representation is close to z.
+
+    the details use seen the notebook (translation_matrix.ipynb)
+
+    >>> transmat = TranslationMatrix(word_pair, source_lang_vec, target_lang_vec)
+    >>> transmat.train(word_pair)
+    >>> translated_word = transmat.translate(words, topn=3)
+
+    """
+    def __init__(self, word_pair, source_lang_vec, target_lang_vec, random_state=None):
+        """
+        Initialize the model from a list pair of `word_pair`. Each word_pair is tupe
+         with source language word and target language word.
+
+        Examples: [("one", "uno"), ("two", "due")]
+
+        Args:
+            `word_pair` (list): a list pair of words
```

Comment:
As a list of many word-pairs, a more natural name for this parameter would be plural  `word_pairs`.

Reply:
fixed